packages/cli/src/environments.ee/source-control/source-control-import.service.ee.ts

```diff
@@ -500,6 +507,30 @@ export class SourceControlImportService {
 		return result;
 	}
 
+	async deleteWorkflowsNotInWorkfolder(user: User, candidates: SourceControlledFile[]) {
+		for (const candidate of candidates) {
+			await this.workflowService.delete(user, candidate.id);
+		}
+	}
+
+	async deleteCredentialsNotInWorkfolder(user: User, candidates: SourceControlledFile[]) {
+		for (const candidate of candidates) {
+			await this.credentialsService.delete(user, candidate.id);
+		}
+	}
+
+	async deleteVariablesNotInWorkfolder(candidates: SourceControlledFile[]) {
+		for (const candidate of candidates) {
+			await this.variablesService.delete(candidate.id);
+		}
+	}
+
+	async deleteTagsNotInWorkfolder(candidates: SourceControlledFile[]) {
+		for (const candidate of candidates) {
+			await this.tagService.delete(candidate.id);
+		}
+	}
+
```

Comment:
these currently fire multiple read and then delete queries. maybe we should add a `deleteMany` method in all these services, and use that to call a single delete query for each resource type.

Reply:
I can do that. Honestly I avoided this because I don't think this is a hot path and being correct was more important to me than being efficient and those functions will always work and there is no chance that they get out of sync with the `deleteMany` and vice versa. 

What I could do is create a `deleteMany` and implement the `delete` using that, similar to this: [`307ede1` (#12329)](https://github.com/n8n-io/n8n/pull/12329/commits/307ede11b99d0f96b541649466dd607ce438675f)

If you approve I'll do that, let's just agree on what we want before we start to work on it.